### PR TITLE
Convert Tooltip to Function Component and Add Unit Tests

### PR DESCRIPTION
**Pull request by ksmort**
## Description
Update Tooltip so that it is a function component.
Add tests + snapshots.
Updating to a function component seems to have fixed the Samsung focus bug.

## Where to Start
components/Tooltip

### CODE REVIEW

File: packages/riipen-ui/src/components/Tooltip.jsx [hunk 1]
```diff
@@ -8,130 +8,35 @@ import withClasses from "../utils/withClasses";
...
+}) => {
+  const theme = React.useContext(ThemeContext);
+  const [isOpen, setIsOpen] = useState(isControlledByProps ? open : false);
+  const tooltipRootRef = useRef(null);
+
+  useEffect(() => {
+    if (isControlledByProps) {
+      setIsOpen(open);
+    }
+  }, [isControlledByProps, open]);
```
Comment (ksmort) [inline packages/riipen-ui/src/components/Tooltip.jsx:37]: Just update state to open prop as needed. Cleans up the logic down the line a bit.

File: packages/riipen-ui/src/components/Tooltip.jsx [hunk 2]
```diff
@@ -406,97 +311,66 @@ class Tooltip extends React.Component {
...
-  getOppositeDirection = (direction1, direction2, value) => {
-    if (value === direction1) {
-      return direction2;
-    } else if (value === direction2) {
-      return direction1;
+  const getOppositeDirection = (direction, oppositeDirection, value) => {
+    if (value === direction) {
+      return oppositeDirection;
+    } else if (value === oppositeDirection) {
+      return direction;
```
Comment (ksmort) [inline packages/riipen-ui/src/components/Tooltip.jsx:318]: Better names. Tbh I still have no idea what past-me was doing here

File: packages/riipen-ui/src/components/Tooltip.jsx [hunk 3]
```diff
@@ -406,97 +311,66 @@ class Tooltip extends React.Component {
...
     }
     return value;
   };
 
-  blur = () => {
+  const blur = () => {
```
Comment (ksmort) [inline packages/riipen-ui/src/components/Tooltip.jsx:323]: I think if there is a focus error again, this is the perpetrator. But it seems okay as a function component?

**Review by jordanell — APPROVED** [review]
All looks good to me!

**Review by stredger — APPROVED** [review]

**Pull request merged**